Add some guidance on becoming a maintainer (#128)
strimzi/strimzi.github.io: contributing/index.md

So as to be transparant and to ensure that all potential maintainers are judged fairly and consistently the following criteria should be taken into account in electing new maintainers:

## Criteria
Some evidence of the following are given particular weight:

* Sustained contributions over at least 3 months ideally including non-trivial PRs.
* An area of developing expertise. This includes the HTTP Bridge and Oauth components. The candidate does not have to be the _go-to_ expert, but they should have a more than superficial understanding of at least one area. For example:
    - A particular operator
    - Some Kubernetes concern (CRDs, API, JSON Schema, and so on)
    - Some associated bit of Kafka technology in which the project has a strategic interest (Cruise Control, for example)
    - System tests
    - Documentation
* Reviewing other contributors' PRs (not just for trivial errors, such as typos, but also guiding contributors on "big picture" aspects such as the future direction of the project).

These criteria are also beneficial: 

> 

Reviewer: ```suggestion
```

* [#strimzi channel on CNCF Slack](https://slack.cncf.io/)
* [Strimzi Dev mailing list](https://lists.cncf.io/g/cncf-strimzi-dev/topics)

# How to become a maintainer

The governance of the project is defined in the [GOVERNANCE.md](https://github.com/strimzi/strimzi-kafka-operator/blob/master/GOVERNANCE.md) file in the Strimzi Github repository, but in summary certain members of the community are the "maintainers" who decide the direction of the project. New maintainers can be elected by a ⅔ majority vote of the existing maintainers.

 So as to be transparant and to ensure that all potential maintainers are judged fairly and consistently the following criteria should be taken into account in electing new maintainers:

> ## Mandatory criteria

Reviewer: ```suggestion
## Considered criteria
```

> Evidence of _all_ of the following are required:

Reviewer: ```suggestion
Following criteria will be considered:
```

* Sustained contributions over at least 3 months including at least two non-trivial PRs.
* An area of developing expertise. The candidate does not have to be the _go-to_ expert, but they should have a more than superficial understanding of at least one area. For example:
    - A particular operator
    - Some Kubernetes concern (CRDs, API, JSON Schema etc).
    - Some associated bit of Kafka technology in which the project has a strategic interest (e.g. Cruise Control)

>     - System tests

Reviewer: ```suggestion
    - System tests
    - Documentation
```

Reviewer 2: It would be good adding related Strimzi components like OAuth library and HTTP bridge as well.

> ## Additional criteria

Reviewer: I would probably remove the title and the followup sentence and merge the two lists into one.

None of these are _strictly_ required, but they're definitely beneficial: 

> * Contributing to multiple aspects of the project and community. For example it’s not enough to just implement features. We should also consider:

Reviewer: ```suggestion
* Contributing to different aspects of the project and community. For example it’s not enough to just implement features. We should also consider:
```

    - bug reporting and fixing
    - Addressing technical debt
    - Documentation

>     - helping users on Slack, the mailing list, stackoverflow etc.

Reviewer: ```suggestion
    - helping users on Slack, GitHub, the mailing lists, stackoverflow etc.
```

    - Promotion of Strimzi (e.g. blogging, conferences etc.)
* An understanding of the overall architecture/structure of the codebase.

> * Knowing when to ask for help or seek consensus.

Reviewer 2: What does it mean exactly?

* An indication of being committed to the long term success of the project.

> ## Examples

Reviewer: Honestly, I would probably remove these.
